content/en/docs/reference/kubernetes-api/ref/common-definitions/podaffinity-.md

```diff
@@ -0,0 +1,73 @@
+---
+api_metadata:
+  apiVersion: ""
+  import: "k8s.io/api/core/v1"
+  kind: "PodAffinity"
+collapsible: false
+content_type: "api_reference"
+description: "Pod affinity is a group of inter pod affinity scheduling rules."
+draft: false
+title: "PodAffinity"
+weight: 16
+---
+
+
+
+`import "k8s.io/api/core/v1"`
+
+
+## PodAffinity
+
+Pod affinity is a group of inter pod affinity scheduling rules.
+
+
+- **preferredDuringSchedulingIgnoredDuringExecution** ([]WeightedPodAffinityTerm)
+
+  The scheduler will prefer to schedule pods to nodes that satisfy the affinity expressions specified by this field, but it may choose a node that violates one or more of the expressions. The node that is most preferred is the one with the greatest sum of weights, i.e. for each node that meets all of the scheduling requirements (resource request, requiredDuringScheduling affinity expressions, etc.), compute a sum by iterating through the elements of this field and adding "weight" to the sum if the node has pods which matches the corresponding podAffinityTerm; the node(s) with the highest sum are the most preferred.
+
+  *The weights of all of the matched WeightedPodAffinityTerm fields are added per-node to find the most preferred node(s)*
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm** (PodAffinityTerm), required
+
+    Required. A pod affinity term, associated with the corresponding weight.
+
+    *Defines a set of pods (namely those matching the labelSelector relative to the given namespace(s)) that this pod should be co-located (affinity) or not co-located (anti-affinity) with, where co-located is defined as running on a node whose value of the label with key <topologyKey> matches that of any node on which a pod of the set of pods is running*
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.topologyKey** (string), required
+
+    This pod should be co-located (affinity) or not co-located (anti-affinity) with the pods matching the labelSelector in the specified namespaces, where co-located is defined as running on a node whose value of the label with key topologyKey matches that of any node on which any of the selected pods is running. Empty topologyKey is not allowed.
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.labelSelector** (<a href="{{< ref "../common-definitions/labelselector-#labelselector" >}}">LabelSelector</a>)
+
+    A label query over a set of resources, in this case pods.
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.namespaces** ([]string)
+
+    namespaces specifies which namespaces the labelSelector applies to (matches against); null or empty list means "this pod's namespace"
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.weight** (int32), required
+
+    weight associated with matching the corresponding podAffinityTerm, in the range 1-100.
+
+- **requiredDuringSchedulingIgnoredDuringExecution** ([]PodAffinityTerm)
```

Comment:
Well... here we have `PodAffinityTerm` again. Where is it defined? 
Are we hoping the users to remember the lines 36-46, or should we repeat that definition here again?

Reply:
It is defined just below, inline, we can see the fields topologyKey, labelSelector and namespaces.

Note that the phrase in italics (Defines a set of pods (namely those matching...) is the description of the `PodAffinityTerm` definition extracted from the spec. Unfortunately, the description does not begin with "PodAffinityTerm defines a set of pods ..." as it is advised, and that does not help the comprehension.